utils/build-script-impl

```diff
@@ -316,55 +316,74 @@ function num_swift_parallel_lto_link_jobs() {
     esac
 }
 
-function set_deployment_target_based_options() {
+function set_build_options_for_host() {
     llvm_cmake_options=()
     swift_cmake_options=()
     cmark_cmake_options=()
     swiftpm_bootstrap_options=()
 
-    case ${deployment_target} in
-        linux-x86_64)
-            SWIFT_HOST_VARIANT_ARCH="x86_64"
-            ;;
-        linux-armv6)
-            # ARM targets require the gold linker
-            USE_GOLD_LINKER=1
-            SWIFT_HOST_VARIANT_ARCH="armv6"
-            ;;
-        linux-armv7)
-            # ARM targets require the gold linker
-            USE_GOLD_LINKER=1
-            SWIFT_HOST_VARIANT_ARCH="armv7"
-            ;;
-        linux-aarch64)
-            # ARM targets require the gold linker
-            USE_GOLD_LINKER=1
-            SWIFT_HOST_VARIANT_ARCH="aarch64"
-            ;;
+    local host="$1"
+
+    case ${host} in
         freebsd-x86_64)
+            SWIFT_HOST_VARIANT="freebsd"
+            SWIFT_HOST_VARIANT_SDK="FREEBSD"
             SWIFT_HOST_VARIANT_ARCH="x86_64"
             ;;
-        linux-powerpc64)
-            SWIFT_HOST_VARIANT_ARCH="powerpc64"
-            ;;
-        linux-powerpc64le)
-            SWIFT_HOST_VARIANT_ARCH="powerpc64le"
-            ;;
         cygwin-x86_64)
+            SWIFT_HOST_VARIANT="windows"
+            SWIFT_HOST_VARIANT_SDK="CYGWIN"
             SWIFT_HOST_VARIANT_ARCH="x86_64"
             ;;
+        linux-*)
+            SWIFT_HOST_VARIANT="linux"
+            SWIFT_HOST_VARIANT_SDK="LINUX"
+            case ${host} in
+                linux-x86_64)
+                    SWIFT_HOST_VARIANT_ARCH="x86_64"
+                    ;;
+                linux-armv6)
+                    # ARM targets require the gold linker
+                    USE_GOLD_LINKER=1
+                    SWIFT_HOST_VARIANT_ARCH="armv6"
+                    ;;
+                linux-armv7)
+                    SWIFT_HOST_VARIANT_ARCH="armv7"
+                    llvm_host_triple="${SWIFT_HOST_VARIANT_ARCH}-linux-gnueabihf"
```

Comment:
I wanted to emphasise that the LLVM and Swift triples should be exactly the same. I've had issues with an "armv7-linux" Swift triple and "arm-linux" LLVM triple, for example. There is probably a better way to do this, such as a single "triple" setting that gets passed to both LLVM and Swift (we can now set `SWIFT_HOST_TRIPLE` explicitly when building, so that's an option)